diff --git src/whyml/mlw_wp.ml src/whyml/mlw_wp.ml
index 3fb1702..aad3be8 100644
--- src/whyml/mlw_wp.ml
+++ src/whyml/mlw_wp.ml
@@ -2002,3 +2002,7 @@ let wp_rec ~wp env kn th fdl =
       th
   in
   List.fold_left add_one th fdl
+
+let wp_expr env kn th e q xq =
+  let env = mk_env env kn th in
+  wp_expr env e q xq
diff --git src/whyml/mlw_wp.mli src/whyml/mlw_wp.mli
index 61b531b..3b9785e 100644
--- src/whyml/mlw_wp.mli
+++ src/whyml/mlw_wp.mli
@@ -36,6 +36,7 @@ val e_now : expr
 
 val pv_old : pvsymbol
 val remove_old : Term.term -> Term.term
+val remove_at : Term.term -> Term.term
 
 val full_invariant :
   Decl.known_map -> Mlw_decl.known_map -> Term.vsymbol -> ity -> Term.term
@@ -49,4 +50,6 @@ val wp_let:
 val wp_rec:
   wp:bool -> Env.env -> known_map -> theory_uc -> fun_defn list -> theory_uc
 
+val wp_expr:
+  Env.env -> known_map -> theory_uc -> expr -> post -> xpost -> invariant
 
